tests/devices/cryostream/test_cryostream.py

```diff
@@ -0,0 +1,237 @@
+        logging.info(f"Running time step: {time}")
+        device_update = cryostream.update(time, inputs={})
+        time_update: Optional[SimTime] = device_update.call_at
+
+        if time_update is None:
+            time = SimTime(int(time) + int(1e9))
+            continue
+        else:
+            time = time_update
+
+    max_diff = 10
+    margin_of_error = np.array([+max_diff, -max_diff])
+    assert any(
+        (device_update.outputs["temperature"] + margin_of_error) == target_temperature
+    )
+
+
+@pytest.mark.asyncio
+async def test_cryostream_update_end(cryostream: Cryostream):
+    starting_temperature = cryostream.default_temp_shutdown - 100
+    cryostream.gas_temp = starting_temperature
+    await cryostream.end(cryostream.default_ramp_rate)
+    assert cryostream.phase_id == PhaseIds.RAMP.value
+    assert cryostream.gas_flow == 10
+
+    time = SimTime(0)
+    device_update: DeviceUpdate
+    while cryostream.phase_id != PhaseIds.HOLD.value:
+        logging.info(f"Running time step: {time}")
+        device_update = cryostream.update(time, inputs={})
+        time_update: Optional[SimTime] = device_update.call_at
+
+        if time_update is None:
+            continue
+        else:
+            time = time_update
+
+    max_diff = 10
+    margin_of_error = np.array([+max_diff, -max_diff])
+    assert any(
+        device_update.outputs["temperature"] + margin_of_error
+        == cryostream.default_temp_shutdown
+    )
+
+
+@pytest.mark.asyncio
+async def test_cryostream_update_plat(cryostream: Cryostream):
+    starting_temperature = cryostream.gas_temp
+    await cryostream.plat(5)
+    assert cryostream.phase_id == PhaseIds.PLAT.value
+
+    time = SimTime(0)
+    time_update: Optional[SimTime] = time
+    device_update: DeviceUpdate
+    while time_update is not None:
+        logging.info(f"Running time step: {time}")
+        device_update = cryostream.update(time, inputs={})
+        time_update = device_update.call_at
+
+        if time_update is None:
+            continue
+        else:
+            time = time_update
+
+    assert device_update.outputs["temperature"] == starting_temperature
+
+
+# # # # # CryostreamAdapter Tests # # # # #
+
+
+@pytest.fixture
+def mock_cryostream() -> Mock:
+    return create_autospec(Cryostream, instance=True)
+
+
+@pytest.fixture
+def raise_interrupt():
+    async def raise_interrupt():
+        return False
+
+    return Mock(raise_interrupt)
+
+
+@pytest.fixture
+def cryostream_adapter(mock_cryostream):
+    return CryostreamAdapter(
+        mock_cryostream, raise_interrupt, host="localhost", port=25565
+    )
+
+
+def test_cryostream_adapter_constructor():
+    CryostreamAdapter(mock_cryostream, raise_interrupt, host="localhost", port=25565)
+
+
+@pytest.mark.asyncio
+async def test_cryostream_adapter_on_connect_gets_device_status(
+    cryostream_adapter: CryostreamAdapter,
+):
+    await cryostream_adapter.on_connect().__anext__()
```

Comment:
`on_connect` must be awaited
```suggestion
    await (await cryostream_adapter.on_connect()).__anext__()
```

Reply:
This change breaks the test. I'll need more detail on why the original is wrong if I'm going to have another stab at getting the test working the way you want it too.